### ISSUE TITLE
Often during changing theme: Exception - Width and Height must be non-negative…

### ISSUE THREAD

**Issue by olegolegoleg** [dremin/RetroBar#661]
During changing themes I often got an expection: `System.ArgumentException: Width and Height must be non-negative. HResult: -2147024809 ParamName: NULL` – [exception.log](https://github.com/dremin/RetroBar/files/12395178/exception.log)

```
[Date: 21.08.2023 12:39:57 Ver: 4.3.0.0]
System.ArgumentException: Width and Height must be non-negative. HResult: -2147024809 ParamName: NULL
   at System.Windows.Rect..ctor(Double x, Double y, Double width, Double height)
   at ScreenFrame.WindowHelper.RECT.op_Implicit(RECT rect)
   at ScreenFrame.WindowHelper.TryGetTaskbar(Rect& taskbarRect, TaskbarAlignment& taskbarAlignment, Boolean& isShown)
   at ScreenFrame.Movers.StickWindowMover.TryGetAdjacentLocationToTaskbar(Double windowWidth, Double windowHeight, Rect& location)
   at ScreenFrame.Movers.StickWindowMover.TryGetAdjacentLocation(Double windowWidth, Double windowHeight, Rect& location)
   at ScreenFrame.Movers.BasicWindowMover.HandleWindowPosChanging(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at ScreenFrame.Movers.WindowMover.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

Often after restart I cannot fully maximize the window:
![Taskbar gap problem](https://github.com/dremin/RetroBar/assets/7686755/7d449a4e-20d8-4ec8-98e0-c4f3c2bd4682)
(this apply to all windows, not just Notepad)

Several restarts of Retrobar helps to get full screen size usable too. 
Also this "space gap" problem often happens during regular start too. 

My screen resolution is 3440x1440.

**Comment by 111ebobo** [comment]
> Hi @111ebobo is this an issue? What is the reason for this?

About the gap I'm pretty sure that disabled auto hide causes the gap issue

**Comment by dremin** [comment]
Regarding the gap: Recent Windows 11 updates have impacted RetroBar's ability to hide the default Windows taskbar. Before starting RetroBar, it is recommended to open Windows Settings > Personalization > Taskbar, select "Taskbar behaviors", then enable "Automatically hide the taskbar."

Regarding the exception: I took a look and it appears the exception is being generated by a program other than RetroBar. Do you know what program may be causing this?

**Comment by dremin** [comment]
> @dremin Autohide helps to fix the gap, thanks!
> 
> 
> 
> But I cannot track any program that causes this exception. Is it possible to enable extended logging or similar?

Does a window appear with that exception? If so you should be able to open Task Manager (while the exception window is open) and see which program the window is under.

**Comment by olegolegoleg** [comment]
I cannot reproduce it with same window, I just got it with Visual Studio Code in the background. New error text:
![Screenshot 2023-08-25 120151-B](https://github.com/dremin/RetroBar/assets/7686755/edd0d1a7-ae0d-4031-8a2d-db1252dc0ed4)

Here details for easier reading:
![image](https://github.com/dremin/RetroBar/assets/7686755/b7bea152-c94f-4079-aff0-2e2e45edaab6)

What is even more important, just before error happens I have not all the themes listed in the drop-down menu! I am sure the red ones were missing:
![image](https://github.com/dremin/RetroBar/assets/7686755/e77a44c6-24aa-47ad-b267-b66c5c7967c3)

I have "Ransomware protection" of the "Microsoft Virus & thread protection" enabled. Since now it is file not found exception, can it be somehow connected?
![image](https://github.com/dremin/RetroBar/assets/7686755/218e6ad7-7395-47bd-b5c7-8f773c844b71)
RetroBar is no in the protected directory, there is only user's documents, desktop, and other disc (%TEMP% is not protected of course).

**Comment by dremin** [comment]
This does seem like RetroBar failed to extract its contents properly. First, close RetroBar, then go to `%temp%\.net` and delete the RetroBar directory. Then, try downloading and running RetroBar again, maybe from a different location this time.

**Comment by olegolegoleg** [comment]
Thanks @dremin,  but still got similar error. This time in the moment when clicked `Properties`:
![image](https://github.com/dremin/RetroBar/assets/7686755/4f4a0948-13a9-4ecc-ae6b-b533657a76f6)

I am using portable version of the RetroBar, that is located at OneDrive: `C:\Users\Oleg\OneDrive\Programs\RetroBar - taskbar customization\RetroBar.exe`

---
I think may it be the `Storage Sense` feature of the Windows 11 that cleans up temporary files even day on my machine:
![image](https://github.com/dremin/RetroBar/assets/7686755/4ac9671a-6f04-4798-90d9-337e53fed0bf)

I will disable it for 1-2 weeks and check whether it may be connected

**Comment by olegolegoleg** [comment]
The disabled Storage Sense (automatic cleanup of temp files) doesn't help.

My next test is RetroBar on disk `D:\RetroBar.exe` (root of the disk), also before test I've removed the RetroBar in `%TEMP%/.net` directory.

Let's hope this will help, I will ping here in 1-2 weeks

**Comment by dremin** [comment]
Thanks for the testing! Do you have any security software installed other than the Microsoft one?

**Comment by olegolegoleg** [comment]
Hi @dremin! No the Microsoft default's "Virus & threat protection" is the only I have. 

The only unusual settings, as I mentioned before, is "Ransomware protection" enabled in "Virus & threat protection". 

This extra security setting works with white list of programs, if a program is in that list it allow it to access the `%USERPROFILE%\Documents` and similar user's directories. 
But the entire `%TEMP%` is definitely not part of the controlled area. 

If I get a new error again, I will test "Ransomware protection" disabled. 
From other possible software I have "Microsoft PowerToys" installed, it contains multiple tweaks and in theory can have some effect on taskbar.

**Comment by olegolegoleg** [comment]
Just got error again:
![image](https://github.com/dremin/RetroBar/assets/7686755/4a62a7de-f467-4675-a308-ffd7416e77e0)

The RetroBar was on the root of the disk, as `D:\RetroBar.exe`. So the location is definitely not the case.

My next test is with disabled "Ransomware protection" (please see comment above).

**Comment by xoascf** [comment]
@olegolegoleg, if something is cleaning up your temp directory, #640 will prevent this message from appearing, as you will not need to have the paths created.
If you want to add languages or themes, you can create the "Languages" and "Themes" folders in "%localappdata%/RetroBar" and place them accordingly.

Try this [build](https://nightly.link/dremin/RetroBar/actions/artifacts/921661352.zip).

**Comment by olegolegoleg** [comment]
Just got error again:
![image](https://github.com/dremin/RetroBar/assets/7686755/4a62a7de-f467-4675-a308-ffd7416e77e0)

The RetroBar was on the root of the disk, as `D:\RetroBar.exe`. So the location is definitely not the case.

Also just finished the 2 weeks  test with disabled "Ransomware protection" (please see comment above), but still getting this error.

Unfortunately, I didn't tracked the source of the issue.
